### PR TITLE
PHOENIX-6282 : Generate PB files inline with build and remove checked in files

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m  8s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  12m 32s |  master passed  |
| +1 :green_heart: |  compile  |   1m 47s |  master passed  |
| +1 :green_heart: |  checkstyle  |   7m 20s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 22s |  master passed  |
| +0 :ok: |  spotbugs  |   4m  1s |  phoenix-core in master has 973 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   5m 36s |  root in master has 1022 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 36s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 34s |  root in the patch failed.  |
| -1 :x: |  cc  |   0m 34s |  root in the patch failed.  |
| -1 :x: |  javac  |   0m 34s |  root in the patch failed.  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  root: The patch generated 0 new + 0 unchanged - 19283 fixed = 0 total (was 19283)  |
| -1 :x: |  prototool  |   0m  6s |  The patch generated 24 new + 6 unchanged - 0 fixed = 30 total (was 6)  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  The patch generated 0 new + 0 unchanged - 9 fixed = 0 total (was 9)  |
| +1 :green_heart: |  shelldocs  |   0m  1s |  There were no new shelldocs issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 4 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  4s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 18s |  phoenix-core in the patch failed.  |
| -1 :x: |  javadoc  |   0m 34s |  root in the patch failed.  |
| -1 :x: |  spotbugs  |   0m 16s |  phoenix-core in the patch failed.  |
| -1 :x: |  spotbugs  |   0m 59s |  root in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 37s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 22s |  The patch generated 1 ASF License warnings.  |
|  |   |  50m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1035 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile spotbugs hbaseanti checkstyle cc prototool shellcheck shelldocs |
| uname | Linux 84f4f3eb9493 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 2b68e84 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| cc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| prototool | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/diff-patch-prototool.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/whitespace-eol.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/whitespace-tabs.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-javadoc-phoenix-core.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-spotbugs-phoenix-core.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-spotbugs-root.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 94 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.7.0 spotbugs=4.1.3 prototool=1.10.0-dev |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by stoty — CHANGES_REQUESTED** [review]
This doesn't seem to find the protoc executable on the Linux CI system.
Adding the protoc artifact logic from OMID-120 should make this multiplatform (or at least get it running on both MacOS and Linux, we only care about those anyway)

**Comment by virajjasani** [comment]
Agree @stoty , Just committing the changes. I missed those changes because locally I already had `protoc` preinstalled.
Thank you

**Comment by virajjasani** [comment]
Thanks for the reference @stoty . I just looked into OMID-120 and confirmed that changes are matching.
At least changes should work for both Linux and MacOS.

File: phoenix-core/pom.xml [hunk 1]
```diff
@@ -194,6 +194,24 @@
...
+        <artifactId>protobuf-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>compile-protoc</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <protoSourceRoot>${basedir}/src/main/protobuf/</protoSourceRoot>
```
Comment (stoty) [inline phoenix-core/pom.xml:208]: These are redundant with  the parent dependencyManagement section.
Setting them here, or in the dependencyManagement section should be enough.

File: pom.xml [hunk 2]
```diff
@@ -1249,6 +1270,11 @@
         <artifactId>protobuf-java</artifactId>
         <version>${protobuf-java.version}</version>
       </dependency>
+      <dependency>
```
Comment (stoty) [inline pom.xml:1273]: I this dependency necessary ?

**Review by stoty — CHANGES_REQUESTED** [review]

File: phoenix-core/pom.xml (hunk 1 above)
Comment (stoty) [reply phoenix-core/pom.xml:210]: Sorry, I only meant the configuration entries.
Declaring the plugin and the version in the top dependencyManagement section is good practice.

Comment (virajjasani) [reply phoenix-core/pom.xml:210]: Kept version declaration property in parent pom and defined entire plugin in phoenix-core, but if declaring plugin with version on parent pom is good practice, let me change it.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 52s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  10m  5s |  master passed  |
| +1 :green_heart: |  compile  |   1m 17s |  master passed  |
| +1 :green_heart: |  checkstyle  |   8m 47s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 54s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 55s |  phoenix-core in master has 973 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   3m 51s |  root in master has 1022 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 31s |  root: The patch generated 0 new + 0 unchanged - 19283 fixed = 0 total (was 19283)  |
| -1 :x: |  prototool  |   0m  3s |  The patch generated 30 new + 6 unchanged - 0 fixed = 36 total (was 6)  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  The patch generated 0 new + 0 unchanged - 9 fixed = 0 total (was 9)  |
| +1 :green_heart: |  shelldocs  |   0m  0s |  There were no new shelldocs issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 4 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m 55s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m  7s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 108m  7s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 57s |  The patch generated 1 ASF License warnings.  |
|  |   | 165m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1035 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile spotbugs hbaseanti checkstyle cc prototool shellcheck shelldocs |
| uname | Linux 422996c090bb 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 2b68e84 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| prototool | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/diff-patch-prototool.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/whitespace-eol.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/whitespace-tabs.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 6598 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.7.0 spotbugs=4.1.3 prototool=1.10.0-dev |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by stoty — COMMENTED** [review]
This looks fine.
Can you upload a PR for 4.x while we wait for the test run ?

**Comment by virajjasani** [comment]
> This looks fine.
> Can you upload a PR for 4.x while we wait for the test run ?

Sure thing. Thanks

**Comment by virajjasani** [comment]
Resolving prototools warnings as separate Jira would be great. Let me try to address few whitespace warnings with this PR.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  1s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 53s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m 53s |  master passed  |
| +1 :green_heart: |  compile  |   1m 19s |  master passed  |
| +1 :green_heart: |  checkstyle  |   8m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 55s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 59s |  phoenix-core in master has 973 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   3m 54s |  root in master has 1022 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  root: The patch generated 0 new + 0 unchanged - 19283 fixed = 0 total (was 19283)  |
| -1 :x: |  prototool  |   0m  1s |  The patch generated 30 new + 6 unchanged - 0 fixed = 36 total (was 6)  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  The patch generated 0 new + 0 unchanged - 9 fixed = 0 total (was 9)  |
| +1 :green_heart: |  shelldocs  |   0m  2s |  There were no new shelldocs issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m  0s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 118m 21s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 57s |  The patch generated 1 ASF License warnings.  |
|  |   | 176m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1035 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile spotbugs hbaseanti checkstyle cc prototool shellcheck shelldocs |
| uname | Linux 633178ce45f9 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 1189b81 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| prototool | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/artifact/yetus-general-check/output/diff-patch-prototool.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 6733 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1035/5/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.7.0 spotbugs=4.1.3 prototool=1.10.0-dev |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by stoty — APPROVED** [review]

**Pull request closed without merging**